### ISSUE TITLE
default implementation for the methods of interface caused System.TypeLoadException

### ISSUE THREAD

**Issue by kinglionsoft** [dotnet/runtime#50989]
### Description

* I create some default implementations in interface [IRepository](https://github.com/kinglionsoft/TypeLoadExceptionSample/blob/bd7d25b4c610f5a5baab7a6cb0d5c65b0bb6ca8e/ConsoleSample/IRepository.cs#L10), then create an abstract class [RepositoryBase](https://github.com/kinglionsoft/TypeLoadExceptionSample/blob/bd7d25b4c610f5a5baab7a6cb0d5c65b0bb6ca8e/ConsoleSample/RepositoryBase.cs#L10) implementing IRepository. Finally I create an empty class [EfRepositoryBase](https://github.com/kinglionsoft/TypeLoadExceptionSample/blob/bd7d25b4c610f5a5baab7a6cb0d5c65b0bb6ca8e/ConsoleSample/EfRepositoryBase.cs#L3) inherited from [RepositoryBase](https://github.com/kinglionsoft/TypeLoadExceptionSample/blob/bd7d25b4c610f5a5baab7a6cb0d5c65b0bb6ca8e/ConsoleSample/RepositoryBase.cs#L10).

* typeof(EfRepositoryBase<,>) caused TypeLoadException

```c#
 static void Main(string[] args)
  {
      var t = typeof(EfRepositoryBase<,>);
      // or
      // typeof(Program).Assembly.DefinedTypes.ToList();
      /*
       *  System.TypeLoadException:
       *     Method 'Max' in type 'ConsoleSample.EfRepositoryBase`2'
       *     from assembly 'ConsoleSample, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
       *     does not have an implementation.
       */
      Console.WriteLine("Hello World!");
  }
```

* sample code: [https://github.com/kinglionsoft/TypeLoadExceptionSample](https://github.com/kinglionsoft/TypeLoadExceptionSample)

* It works if the abstract class RepositoryBase fully implementing IRepository including the default implementations :
https://github.com/kinglionsoft/TypeLoadExceptionSample/blob/bd7d25b4c610f5a5baab7a6cb0d5c65b0bb6ca8e/ConsoleSample/RepositoryBase.cs#L85

### Configuration
* dotnet --info
```

.NET SDK (反映任何 global.json):
 Version:   5.0.202
 Commit:    db7cc87d51

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.202\

Host (useful for support):
  Version: 5.0.5
  Commit:  2f740adc14
```

**Comment by sgtfrankieboy** [comment]
I think I'm dealing with the same error as you. I managed to get rid of the TypeLoadException by ensuring that the last method of the interface has a default implementation. 

So this worked:
```
public interface IFoo {
    
    int Divide(int x, int y);

    int Multiply(int x, int y);

    int Sum(int x, int y) => x + y;
}
```

and this caused it to throw:
```
public interface IFoo {
    
    int Sum(int x, int y) => x + y;
  
    int Divide(int x, int y);

    int Multiply(int x, int y);
}
```

That interface is being implemented by a `AbstractFoo` which itself was being implemented by `ImplFoo`. The TypeLoadException that is being thrown was saying Method `Multiply` is missing an implementation in type `ImplFoo`.

I also managed to fix it by implementing the methods with a default implementation in AbstractFoo.
